service/history/api/resetworkflow/api.go

```diff
@@ -163,11 +164,46 @@ func Invoke(
 		),
 		request.GetReason(),
 		nil,
-		request.GetResetReapplyType(),
+		GetResetReapplyExcludeTypes(request.GetResetReapplyExcludeTypes(), request.GetResetReapplyType()),
 	); err != nil {
 		return nil, err
 	}
 	return &historyservice.ResetWorkflowExecutionResponse{
 		RunId: resetRunID,
 	}, nil
 }
+
+// GetResetReapplyExcludeTypes computes the set of requested exclude types. It
+// uses the reset_reapply_exclude_types request field (a set of event types to
+// exclude from reapply), as well as the deprecated reset_reapply_type request
+// field (a specification of what to include).
+func GetResetReapplyExcludeTypes(
+	excludeTypes []enumspb.ResetReapplyExcludeType,
+	includeSpec enumspb.ResetReapplyType,
+) []enumspb.ResetReapplyExcludeType {
```

Comment:
nit: feels like excludeSet is easier to work with here and in the resetter?

Reply:
Thanks. I agree, I also wanted to do that for convenience. Just my lack of knowledge about what is idiomatic in Go held me back. I've made that change now.